### ISSUE TITLE
[Linux] Themes are not detected in .ultrastardx/themes

### ISSUE THREAD

**Issue by jmfergeau** [UltraStar-Deluxe/USDX#715]
### Actual behaviour
- Custom themes are not found by USDX, leaving only the choices between Deluxe and Modern, ignoring what's inside the themes folder.
- If a custom theme is forced in the config.ini, it's forced back on the Modern theme.
- No relevant errors are thrown. There's lots of lines about themes being logged but they're also logged when no custom theme is there or set:
```
ERROR:  no texture type for LoadingStaticAnimation found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic1Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic2Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic3Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic4Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic5Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic6Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic7 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic7Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic8 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic8Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic9 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic9Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic10 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic10Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic11 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic11Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic12 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic12Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SongChessboardStaticCat found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for JukeboxTimeProgress found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for JukeboxSongBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for JukeboxSongCover found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for JukeboxStaticActualSongCover found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for JukeboxSongMenuTimeProgress found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingTimeProgress found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1SingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1TwoPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1ThreePSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2RSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2MSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3SingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1TwoPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1ThreePAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2RAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2MAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3RAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP1ThreePStatic2 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP1ThreePAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP2MStatic2 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP2MAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP3RStatic2 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP3RAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP1ThreePSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP2MSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP3RSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1FourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1FourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2FourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2FourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3FourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3FourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4FourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4FourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1SixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1SixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2SixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2SixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3SixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3SixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4SixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4SixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP5SixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP5SixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP6SixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP6SixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1DuetFourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1DuetFourPStatic2 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1DuetFourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2DuetFourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2DuetFourPStatic2 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2DuetFourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3DuetFourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3DuetFourPStatic2 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3DuetFourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4DuetFourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4DuetFourPStatic2 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4DuetFourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1DuetSixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1DuetSixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2DuetSixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2DuetSixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3DuetSixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3DuetSixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4DuetSixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4DuetSixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP5DuetSixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP5DuetSixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP6DuetSixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP6DuetSixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for EditSubHeaderBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for EditSubCurrentNoteInfoBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for EditSubVolumeSliderBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for EditSubNotesBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for EditSubP1InfoBarBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for EditSubP2InfoBarBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for EditSubSentenceBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for ScoreDownloadPopupProgressSong found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for ScoreDownloadPopupProgressWeb found. [TTheme.ThemeLoadStatic]
```

### Expected behaviour
If a theme (ini and folder sharing the same name) is found in the themes folder, it should be selectable in the themes list in options.

### Steps to reproduce
1. Put a custom theme (in the example, it's a copy of modern with a different name) in the `.ultrastardx/themes` folder
2. Run ultrastardx
3. Go to Options then Themes
4. Custom theme doesn't appear in the themes list

### Details
- USDX version: 2023.5 appimage (does it on the flatpak version too)
- Operating System + version: Arch Linux with 6.1.33 kernel

I noticed some people got problems with themes recently. Is this feature locked on purpose?
Couldn't try on Windows or Mac so I don't know if it's a linux only issue or if it's even related on how appimage and flatpak works.

**Comment by s09bQ5** [comment]
Which folders have you tried as "the themes folder" in the AppImage and in the Flatpak case?

**Comment by barbeque-squared** [comment]
You can ignore the logged errors in this context, see #697 for that.

Slightly offtopic, but can confirm that regular Linux also completely ignores the `themes` folder in `~/.ultrastardx`. I have to put mine into `/usr/share/ultrastardx/themes` (and put them there _again_ everytime I install/update, too)

**Comment by s09bQ5** [comment]
> Slightly offtopic, but can confirm that regular Linux also completely ignores the `themes` folder in `~/.ultrastardx`.

You are right, we never supported themes in ~/.ultrastardx/themes. So with the AppImage the only way to use other themes is to unpack the AppImage, e.g.:
```
./UltraStarDeluxe-v2023.5.0-0-x86_64.AppImage --appimage-extract usr
sudo chown -R root:root squashfs-root/usr
sudo mv squashfs-root/usr /opt/UltraStar-Deluxe
rmdir squashfs-root
/opt/UltraStar-Deluxe/bin/ultrastardx
```
Themes can then be installed in /opt/UltraStar-Deluxe/share/ultrastardx/themes.

For the Flatpak the share/ultrastardx/themes directory has been declared as an extension point.

**Comment by barbeque-squared** [comment]
Should we support this? If Mac already does it, I don't see much reason why we can't on Linux. But there's some very real pro/cons to be deliberated:
- pro: easier for people to use custom themes (no root required)
- con: "the game crashes/looks weird because I use a super outdated custom theme!"

the latter should be less of an issue once #697 is implemented imo, and even less if perhaps we can make USDX smart enough that if it detects a theme error it can't recover from, it just resets it back to the default and pops up some error message (that tells you to look in the Error.log).
I'd like those to be in before we start making theme support better.

**Comment by s09bQ5** [comment]
macOS does it by copying everything from the bundle to the user's home directory IIRC.

**Comment by jmfergeau** [comment]
> Which folders have you tried as "the themes folder" in the AppImage and in the Flatpak case?

`~/.ultrastardx/themes` for appimage and `~/.var/app/eu.usdx.UltraStarDeluxe/.ultrastardx/themes` for flatpak

**Comment by s09bQ5** [comment]
Btw., the flatpak defines an extension eu.usdx.UltraStarDeluxe.Themes merged into the directory /app/share/ultrastardx/themes. So you can create a flatpak with that name that contains your custom themes.
